js/ui/tree_view/ui.tree_view.base.js

```diff
@@ -899,9 +958,14 @@ const TreeViewBase = HierarchicalCollectionWidget.inherit({
             }
         }
 
-        const $icon = $node.children('.' + TOGGLE_ITEM_VISIBILITY_CLASS);
+        const $icon = $node.children(`.${TOGGLE_ITEM_VISIBILITY_CLASS},.${CUSTOM_ICON_TOGGLE_ITEM_VISIBILITY_CLASS}`);
+
         const $nodeContainer = $node.children(`.${NODE_CONTAINER_CLASS}`);
 
+        if(this._showCustomExpanderIcons() && $nodeContainer.not(':empty').length) {
+            this._toggleCustomExpanderBtnIcons($node.children(`.${CUSTOM_EXPAND_ICON_CLASS}`), $node.children(`.${CUSTOM_COLLAPSE_ICON_CLASS}`), state);
```

Comment:
I removed this code, run test and found them green :)
Moreover, I tested locally and didn't find any problem.

Why do we need this?

Reply:
it toggle icon, but before this we check if icon needs to be toggled, since in virtual mode we may not know parent has children. Anyway i'll look into this, you right that some test is needed here

Reply:
update: test added